cv32e40s/tests/programs/custom/illegal_access_loop_tests/csr_privilege_gen.py

```diff
@@ -0,0 +1,80 @@
+"""
+script to generate the csr_register_loop
+
+"""
+
+input_string = "// Start of generated code." # start string, this will move the wrie HEAD
+pointer = 0 # file pointer 
+filename = "csr_privilege_loop.S" # file which will be written to 
+
+
+# register list fetched manually from the spec (V20211203), contains all U-, S-, R-, and M-mode CSR registers
+reg_str = """
```

Comment:
Why do we need a list of individual ranges?
Is it not contiguously covering all of 0x000-0xFFF (or 0x100-0xFFF), or is there some differentiation in how the different ranges are tested?

Reply:
From discussions in chat: The reason is to exclude User-Level CSRs, as per the purpose of the test.

It would also be possible to loop over `0x000-0xFFF` and filter out `[9:8] == 0`, but these are just 30 lines so I think it is good as it is now.

Reply:
Should I mark this as resolved as well ?

Reply:
Sure. The answer made sense and the code seems good.